File: src/com/aslan/module/command/ImmediateInvoker.js
import Invoker from "./Invoker";
import Lookup from "../../config/lookup.js";

export default class ImmediateInvoker extends Invoker {

	invoke(command) {
		try {
			command.execute();
		} catch (error) {
			Lookup.notification.notify({name: "command.error", attachment: error});
		}
	}
}

File: src/com/aslan/module/command/ImmediateInvoker.java
package com.aslan.module.command;

import com.aslan.config.Lookup;
import com.aslan.module.notify.NotifyEvent;

public class ImmediateInvoker implements Invoker {

    public void invoke(Command command) {
        try {
            command.execute();
        } catch (Exception e) {
            Lookup.notification().notify(new NotifyEvent("command.error", e));
        }
    }
}